smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -556,66 +546,87 @@ def poll() {
 		unsubscribe()
 		return
 	}
+	def isThermostatPolled = !(thermostats || ecobeesensors) // If no thermostats or sensors, mark them polled
+	def isSwitchesPolled = !(ecobeeswitches)                 // If no switches, mark them polled
+	def pollAttempt = 1
 
-	try{
-		// First check if we need to poll thermostats or sensors
-		if (thermostats || ecobeesensors) {
-			def requestBody = [
-				selection: [
-					selectionType:          "registered",
-					selectionMatch:         "",
-					includeExtendedRuntime: true,
-					includeSettings:        true,
-					includeRuntime:         true,
-					includeSensors:         true
+	while (!(isThermostatPolled && isSwitchesPolled) && (pollAttempt < 3)) {
+		try{
+			// First check if we need to poll thermostats or sensors
+			if (!isThermostatPolled) {
+				def requestBody = [
+					selection: [
+						selectionType:          "registered",
+						selectionMatch:         "",
+						includeExtendedRuntime: true,
+						includeSettings:        true,
+						includeRuntime:         true,
+						includeSensors:         true
+					]
+				]
+				def pollParams = [
+					uri: apiEndpoint,
+					path: "/1/thermostat",
+					headers: ["Content-Type": "text/json", "Authorization": "Bearer ${state.authToken}"],
+					// TODO - the query string below is not consistent with the Ecobee docs:
+					// https://www.ecobee.com/home/developer/api/documentation/v1/operations/get-thermostats.shtml
+					query: [format: 'json', body: toJson(requestBody)]
 				]
-			]
-			def pollParams = [
-				uri: apiEndpoint,
-				path: "/1/thermostat",
-				headers: ["Content-Type": "text/json", "Authorization": "Bearer ${state.authToken}"],
-				// TODO - the query string below is not consistent with the Ecobee docs:
-				// https://www.ecobee.com/home/developer/api/documentation/v1/operations/get-thermostats.shtml
-				query: [format: 'json', body: toJson(requestBody)]
-			]
 
-			httpGet(pollParams) { resp ->
-				if(resp.status == 200) {
-					if (thermostats || ecobeesensors) {
-						storeThermostatData(resp.data.thermostatList)
+				httpGet(pollParams) { resp ->
+					isThermostatPolled = true
+					if(resp.status == 200) {
+						if (thermostats || ecobeesensors) {
+							storeThermostatData(resp.data.thermostatList)
+						}
+						if (ecobeesensors) {
+							updateSensorData(resp.data.thermostatList.remoteSensors)
+						}
```

Comment:
should we add an `else` for other `resp.status` codes like we have for switches below?

Reply:
Other than 200 there shouldn't really be any other successful response statuses. If there wasn't any data updated during the poll, the health check coming in next PR will count this loss of data towards the 3 attempts before marking the device offline.
As log.warn only shows up in live logs they don't give that much information in prod.